slider/beatmap.py

```diff
@@ -479,12 +485,19 @@ def tick_points(self):
         pre_repeat_ticks = []
         append_tick = pre_repeat_ticks.append
 
-        time_between_ticks = repeat_duration / ticks
-        for n in range(1, ticks + 1):
-            pos = curve(n / ticks)
-            timediff = n * time_between_ticks
+        t = 0
+        while True:
+            t += self.tick_rate
+            if t >= self.num_beats:
+                pos = curve(1)
+                timediff = repeat_duration
+                append_tick(Point(int(pos.x), int(pos.y), time + timediff))
```

Comment:
I've honestly forgotten XD. It was a lot of code ago I wrote this bit. I think my train of thought was something like "all osu coordinates must be integers so this should be too" but that's probably crazy.